src/csharp/Microsoft.Spark.Worker.UnitTest/TaskRunnerTests.cs

```diff
@@ -34,69 +34,9 @@ public void TestTaskRunner()
-
-                payloadWriter.Write(outputStream, payload, commandPayload);
-
-                // Write 10 rows to the output stream.
-                var pickler = new Pickler();
-                for (int i = 0; i < 10; ++i)
-                {
-                    var pickled = pickler.dumps(
-                        new[] { new object[] { i.ToString(), i, i } });
-                    SerDe.Write(outputStream, pickled.Length);
-                    SerDe.Write(outputStream, pickled);
-                }
-
-                // Signal the end of data and stream.
-                SerDe.Write(outputStream, (int)SpecialLengths.END_OF_DATA_SECTION);
-                SerDe.Write(outputStream, (int)SpecialLengths.END_OF_STREAM);
-                outputStream.Flush();
-
+                payloadWriter.WriteTestData(outputStream);
                 // Now process the bytes flowing in from the client.
-                var timingDataReceived = false;
-                var exceptionThrown = false;
-                var rowsReceived = new List<object[]>();
-
-                while (true)
-                {
-                    var length = SerDe.ReadInt32(inputStream);
-                    if (length > 0)
-                    {
-                        var pickledBytes = SerDe.ReadBytes(inputStream, length);
-                        using var unpickler = new Unpickler();
-                        var rows = unpickler.loads(pickledBytes) as ArrayList;
-                        foreach (object row in rows)
-                        {
-                            rowsReceived.Add((object[])row);
-                        }
-                    }
-                    else if (length == (int)SpecialLengths.TIMING_DATA)
-                    {
-                        var bootTime = SerDe.ReadInt64(inputStream);
-                        var initTime = SerDe.ReadInt64(inputStream);
-                        var finishTime = SerDe.ReadInt64(inputStream);
-                        var memoryBytesSpilled = SerDe.ReadInt64(inputStream);
-                        var diskBytesSpilled = SerDe.ReadInt64(inputStream);
-                        timingDataReceived = true;
-                    }
-                    else if (length == (int)SpecialLengths.PYTHON_EXCEPTION_THROWN)
-                    {
-                        SerDe.ReadString(inputStream);
-                        exceptionThrown = true;
-                        break;
-                    }
-                    else if (length == (int)SpecialLengths.END_OF_DATA_SECTION)
-                    {
-                        var numAccumulatorUpdates = SerDe.ReadInt32(inputStream);
-                        SerDe.ReadInt32(inputStream);
-                        break;
-                    }
-                }
-
-                Assert.True(timingDataReceived);
-                Assert.False(exceptionThrown);
+                var rowsReceived = PayloadReader.Read(inputStream);
```

Comment:
nit: cannot use `var` here.

Reply:
@samueleresca, this has not been fixed.

Reply:
> @samueleresca, this has not been fixed.
@elvaliuliuliu thanks for the feedback, here is the fix: #425